### PR TITLE
[ADD] partner_email_unique: new module

### PR DESCRIPTION
**Pull request by robinkeunen**
New module, adds an unique constraint to email field.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please include a safeguard for avoiding this interferes with other tests. You have something similar in:

https://github.com/OCA/hr-timesheet/blob/12.0/hr_timesheet_employee_required/models/account_analytic_line.py

**Comment by robinkeunen** [comment]
Hi @pedrobaeza , thank you for the quick review. Something like the last _fixup_ commit?

**Comment by pedrobaeza** [comment]
Not exactly. I have put a comment in the commit itself.

**Comment by robinkeunen** [comment]
Yup, here is another go at it.

**Comment by pedrobaeza** [comment]
Yeah, that one seems good! I let other reviewers full review now.

**Review by houssine78 — APPROVED** [review]
LGTM

**Review by PierrickBrun — APPROVED** [review]

File: partner_email_unique/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from openerp.exceptions import ValidationError
+from openerp.tools import config
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.multi
+    @api.constrains('email')
+    def _check_email(self):
```
Comment (rousseldenis) [inline partner_email_unique/models/res_partner.py:16]: @robinkeunen For constraints like this, you maybe use that defined in https://github.com/OCA/partner-contact/pull/706

It avoids too much code :smile:

Comment (pedrobaeza) [reply partner_email_unique/models/res_partner.py:16]: Please check my comment at https://github.com/OCA/partner-contact/pull/720#issuecomment-475663856

File: partner_email_unique/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "website": "https://odoo-community.org/",
+    "author": "Coop IT Easy SCRL fs, "
+              "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": [
+        "base",
+    ],
+    "data": [
```
Comment (NL66278) [inline partner_email_unique/__openerp__.py]: Remove empty keys.

File: partner_email_unique/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from . import res_partner
```
Comment (NL66278) [inline partner_email_unique/models/__init__.py:6]: Missing coding and license.

File: partner_email_unique/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (NL66278) [inline partner_email_unique/__openerp__.py]: License lines should be https

File: partner_email_unique/tests/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+from . import test_res_partner_email
```
Comment (NL66278) [inline partner_email_unique/tests/__init__.py:6]: Missing coding and license here as well

**Review by NL66278 — CHANGES_REQUESTED** [review]
Few comments related to current OCA conventions. Nothing really blocking, but if possible to change, it would be nice.

**Comment by robinkeunen** [comment]
@NL66278 No worries, here you go.

File: partner_email_unique/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Partner unique email",
+    "summary": "Add an unique constraint to email field",
+    "version": "9.0.1.0.0",
+    "category": "Customer Relationship Management",
+    "website": "https://odoo-community.org/",
```
Comment (NL66278) [inline partner_email_unique/__openerp__.py]: I missed this one on the review, sorry: website should be: ```https://github.com/oca/partner-contact```

File: partner_email_unique/tests/__init__.py (hunk 5 above)
Comment (NL66278) [reply partner_email_unique/tests/__init__.py:6]: You forgot this one I think?

File: partner_email_unique/__openerp__.py (hunk 6 above)
Comment (robinkeunen) [reply partner_email_unique/__openerp__.py]: 👌  No worries

**Review by NL66278 — APPROVED** [review]

**Comment by robinkeunen** [comment]
Thanks! First merge into the OCA \o/

But you merged the fixup commit without squashing it, a little rebase would make it cleaner. :)

**Pull request merged**